### PR TITLE
SLCORE-339 Allow to pass extra properties in telemetry

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/telemetry/payload/TelemetryPayload.java [hunk 1]
```diff
@@ -79,12 +84,14 @@
   private final TaintVulnerabilitiesPayload taintVulnerabilitiesPayload;
 
   @SerializedName("rules")
-  private TelemetryRulesPayload telemetryRulesPayload;
+  private final TelemetryRulesPayload telemetryRulesPayload;
+
+  private transient final Map<String, Object> additionalAttributes;
```
Comment (jblievremont) [inline core/src/main/java/org/sonarsource/sonarlint/core/telemetry/payload/TelemetryPayload.java:89]: Nitpick (and flagged on Next): modifiers are not in JLS-recommended order

File: core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryClientAttributesProvider.java [hunk 2]
```diff
@@ -58,4 +59,9 @@
    */
   Collection<String> getDefaultDisabledRules();
 
+  /**
+   * Map of additional attributes to be passed to the telemetry. Values types can be {@link String}, {@link Boolean} or {@link Number}. You can also pass a Map for nested objects.
```
Comment (jblievremont) [inline core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryClientAttributesProvider.java:63]: 👍🏻

**Review by jblievremont — APPROVED** [review]
LGTM 👍🏻

File: core/src/main/java/org/sonarsource/sonarlint/core/telemetry/payload/TelemetryPayload.java (hunk 1 above)
Comment (henryju) [reply core/src/main/java/org/sonarsource/sonarlint/core/telemetry/payload/TelemetryPayload.java]: Fixed

**Pull request merged**